examples/with-typescript/tsconfig.json

```diff
@@ -1,8 +1,12 @@
 {
     "compilerOptions": {
-        "module": "commonjs",
-        "target": "es2015",
-        "jsx": "react",
-        "allowJs": true
+        "module": "es2015",
+        "target": "es2017",
```

Comment:
why are you changing these two fields? this is a pretty significant shift in how the example works. what's wrong with commonjs es6?